### PR TITLE
With http network logs

### PR DESCRIPTION
**Pull request by maskaravivek**
## Title (required)

Adds request logs to help in debugging #1485.

## Description (required)

Other options were not working well for me, so I have written a small implementation of http interceptor to help debug network traffic.

## Tests performed (required)

Have tested that the logs are getting printed properly.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/mwapi/ApacheHttpClientMediaWikiApi.java [hunk 1]
```diff
@@ -85,6 +85,7 @@ public ApacheHttpClientMediaWikiApi(Context context,
         ClientConnectionManager cm = new ThreadSafeClientConnManager(params, schemeRegistry);
         params.setParameter(CoreProtocolPNames.USER_AGENT, getUserAgent());
         httpClient = new DefaultHttpClient(cm, params);
+        httpClient.addRequestInterceptor(NetworkInterceptors.getHttpRequestInterceptor());
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/mwapi/ApacheHttpClientMediaWikiApi.java]: Nice! :-)

File: app/src/main/java/fr/free/nrw/commons/mwapi/NetworkInterceptors.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import timber.log.Timber;
+
+public class NetworkInterceptors {
+
+    /**
+     * Interceptor to log the HTTP request
+     */
+    @NonNull
+    public static HttpRequestInterceptor getHttpRequestInterceptor() {
+        return (HttpRequest request, HttpContext httpContext) -> {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/mwapi/NetworkInterceptors.java]: To minimize impact on performance, how about this:
- Check whether the app in in prod or not
- If prod, do nothing. If not, do the logging
? In the spirit of rule 4 at https://gualtierotesta.wordpress.com/2015/11/29/the-5-java-logging-rules/
Or we can leave it on for everybody until we don't have anymore network related bugs, and maybe do that check from 2.9.

Comment (maskaravivek) [reply app/src/main/java/fr/free/nrw/commons/mwapi/NetworkInterceptors.java]: Yes, for now, we can probably log both for debug and release builds. Later on, we can consider running it just on debug builds.

**Comment by nicolas-raoul** [comment]
I am testing this PR.
It can sometimes be hard to distinguish requests, for instance:
```
07-27 15:06:18.286 14776 16009 V NetworkInterceptors: <<<<<<<<<<<<<< REQUEST PARAMS >>>>>>>>>>>>
07-27 15:06:18.287 14776 16009 V NetworkInterceptors: <<<<<<<<<<<<<< REQUEST PARAMS >>>>>>>>>>>>
07-27 15:06:18.287 14776 16009 V NetworkInterceptors: <<<<<<<<<<<<<< HEADERS >>>>>>>>>>>>
07-27 15:06:18.288 14776 16009 V NetworkInterceptors: Header >> Host: commons.wikimedia.org
07-27 15:06:18.289 14776 16009 V NetworkInterceptors: Header >> Connection: Keep-Alive
07-27 15:06:18.289 14776 16009 V NetworkInterceptors: Header >> User-Agent: Commons/2.7.2-debug-2.8-release~dac91fe8 (https://mediawiki.org/wiki/Apps/Commons) Android/7.1.2
07-27 15:06:18.290 14776 16009 V NetworkInterceptors: Header >> Cookie: WMF-Last-Access=27-Jul-2018; GeoIP=JP:13:Tokyo:35.69:139.75:v4; commonswikiSession=11nurhmjs788gjadnb1f932e3mc587a3; commonswikiUserID=27103; commonswikiUserName=Syced; forceHTTPS=true; centralauth_User=Syced; centralauth_Token=224c040e9073791df07640eaa0140d63; centralauth_Session=e4aeb162d4bf137f353d04e478784190; loginnotify_prevlogins=2018-1ukl9cq-gubwfiefmqodwg5k0vz9xku03u3aw8d
07-27 15:06:18.290 14776 16009 V NetworkInterceptors: Header >> Cookie2: $Version=1
07-27 15:06:18.291 14776 16009 V NetworkInterceptors: <<<<<<<<<<<<<< HEADERS >>>>>>>>>>>>
07-27 15:06:18.291 14776 16009 V NetworkInterceptors: Protocol version:
07-27 15:06:18.291 14776 16009 V NetworkInterceptors:  HTTP/1.1
```
But I guess we will be able to distinguish them when needed, so merging for now.

**Comment by maskaravivek** [comment]
@nicolas-raoul I added the following logs to distinguish between the requests. 

```
<<<<<<<<<<<<<< START OF REQUEST LOGGING 
<<<<<<<<<<<<<< END OF REQUEST LOGGING 
```

Is something else required?

**Comment by misaochan** [comment]
This PR works amazingly (thanks @maskaravivek !), but we need to decide whether we should remove this logging before the release (or implement a check for debug build). Releasing this with the logging in place would have privacy concerns IMO, especially since coordinates and IP are logged.

**Comment by nicolas-raoul** [comment]
It is only the phone's logs, so privacy is not such a big issue. For instance I have seen a web browser log each page I was visiting.

**Comment by misaochan** [comment]
These headers would be incorporated in the logs that users send to us if the Send Logs button worked, though. Given WMF's stand on privacy, I guess it may be safer to at least modify the Settings subtext for "send logs" to mention that the logs may include IP and coordinates?

**Pull request merged**